script.js

```diff
@@ -1,28 +1,98 @@
+      episodeLength = allEpisodes.length;
+      makeSearchSection();
+      fillSelectByEpisodes(allEpisodes);
+      makeSectionForEpisodes(allEpisodes);
+    })
+    .catch((error) => console.error(error));
 }
-function createSelectEpisode() {
+
+function makeSectionForSelects() {
+  const rootElem = document.getElementById("root");
   let searchSection = document.createElement("section");
-  searchSection.id = "searchBox";
-  searchSection.className = "row p-4 ";
-  let SelectElement = document.createElement("select");
-  SelectElement.className = "col-4 m-3 p-2 form-control";
-  SelectElement.id = "selectEpisode";
-  let allEpisodes = getAllEpisodes();
+  searchSection.id = "searchSection";
+  searchSection.className = "row p-4 mx-auto";
+  rootElem.append(searchSection);
+  let showSelectElement = document.createElement("select");
+  showSelectElement.id = "showSelect";
+  showSelectElement.className = "col-10 mx-auto p-2";
+  let episodeSelectElement = document.createElement("select");
+  episodeSelectElement.className = "col-3 m-3 p-2 form-control";
+  episodeSelectElement.id = "selectEpisode";
+  searchSection.append(
+    showSelectElement,
+    episodeSelectElement
+  );
+}
+
+function fillSelectByShows(selectedShow) {
+  let searchSection = document.getElementById("searchSection");
+  let SelectElement = document.getElementById("showSelect");
+  let allShows = getAllShows();
+  allShows.map((show) => {
+    let optionText = show.name;
+    let optionElement = document.createElement("option");
+    optionElement.innerText = show.name;
+    optionElement.value = show.id;
+    SelectElement.appendChild(optionElement);
+  });
+  SelectElement.selectedIndex = selectedShow;
+  SelectElement.addEventListener("change", function () {
+    let url = `https://api.tvmaze.com/shows/${this.value}/episodes`;
+    let rootElem = document.getElementById("root");
+    rootElem.innerHTML = ``;
+    setup(url, this.selectedIndex);
+  });
+}
+
+function makeSearchSection() {
+  let searchSection = document.getElementById("searchSection");
+  let inputElement = document.createElement("input");
+  inputElement.id = "searchKey";
+  inputElement.className = "col-3 m-3  form-control";
+  let labelElement = document.createElement("label");
+  labelElement.id = "searchComment";
+  labelElement.className = "col-4 m-3 text-center";
+  searchSection.append(inputElement, labelElement);
+  inputElement.addEventListener("input", search);
+}
+
+function search() {
+  let episodeFilter = allEpisodes.filter(filterByKey);
+  const rootElem = document.getElementById("root");
+  let container = document.getElementById("container");
+  rootElem.removeChild(container);
+  makeSectionForEpisodes(episodeFilter);
+}
+
+function filterByKey(episode) {
+  let searchKey = document.getElementById("searchKey").value;
+  return (
+    episode.name.toLowerCase().includes(searchKey.toLowerCase()) ||
+    episode.summary.toLowerCase().includes(searchKey.toLowerCase())
```

Comment:
@DibaKarimi  you can create two variables and instead of these long codes call those variables